infra/infra.go

```diff
@@ -42,8 +43,8 @@ func NewDefaultProvider(cluster *v1.Cluster) Interface {
 		if err != nil {
 			logger.Error(err)
 		}
-		return aliProvider
+		return aliProvider, nil
 	default:
-		return nil
+		return nil, fmt.Errorf("the provider is invalid")
```

Comment:
as my opinion, NewDefaultProvider is just a New method for struct ,shoud not return error, nil means New struct failed.

Reply:
Sometimes it will failed, like can't found AK